torchrl/modules/models/multiagent.py

```diff
@@ -54,9 +57,11 @@ def __init__(
         self.__dict__["_empty_net"] = self._build_single_net(**kwargs)
 
     @property
-    def _vmap_randomness(self):
+    def vmap_randomness(self):
         if self.initialized:
-            return "error"
+            return self._vmap_randomness
+        # Matteo: There seems to be a problem with lazy layers when using "different" here
```

Comment:
taking care of this.
We should avoid leaving comments with names like this, it's very easy to miss during review and can stay there forever!

Reply:
This is the culprit:
https://github.com/pytorch/tensordict/blob/f09f20dd586ee1749d244e5ae42ee5cdc992bd77/tensordict/_td.py#L771-L776

When we call vmap we actually don't batch the lazy tensor (we can't do it) so it must be `"same"`. `"same"` isn't a problem though because the tensor isn't batched so it's not going to copy the same config for every element. TLDR we can stick to same

Reply:
Ok then can you write  a comment with this in the code? and then we are good to go!